include/secp256k1_musig.h

```diff
@@ -7,348 +7,152 @@
 extern "C" {
 #endif
 
-#include <stdint.h>
+#include <stddef.h>
 
-/** This module implements a Schnorr-based multi-signature scheme called MuSig
- * (https://eprint.iacr.org/2018/068.pdf). It is compatible with bip-schnorr.
+/** This module implements a Schnorr-based multi-signature scheme called MuSig2
+ * (https://eprint.iacr.org/2020/1261, see Appendix B for the exact variant).
+ * Signatures are compatible with BIP-340 ("Schnorr").
  * There's an example C source file in the module's directory
  * (src/modules/musig/example.c) that demonstrates how it can be used.
  *
- * The documentation in this include file is for reference and may not be sufficient
- * for users to begin using the library. A full description of API usage can be found
- * in src/modules/musig/musig.md
+ * The module also supports BIP-341 ("Taproot") public key tweaking and adaptor
+ * signatures as described in
+ * https://github.com/ElementsProject/scriptless-scripts/pull/24
+ *
+ * It is recommended to read the documentation in this include file carefully.
+ * Further notes on API usage can be found in src/modules/musig/musig.md
+ *
+ * You may know that the MuSig2 scheme uses two "nonces" instead of one. This
+ * is not wrong, but only a technical detail we don't want to bother the user
+ * with. Therefore, the API only uses to the singular term "nonce".
+ *
+ * Since the first version of MuSig is essentially replaced by MuSig2, when
+ * writing MuSig or musig here we mean MuSig2.
  */
 
-/** Data structure containing auxiliary data generated in `pubkey_combine` and
- *  required for `session_*_init`.
- *  Fields:
- *        magic: Set during initialization in `pubkey_combine` to allow
- *               detecting an uninitialized object.
- *      pk_hash: The 32-byte hash of the original public keys
- *    second_pk: Serialized x-coordinate of the second public key in the list.
- *               Filled with zeros if there is none.
- *    pk_parity: Whether the MuSig-aggregated point was negated when
- *               converting it to the combined xonly pubkey.
- *     is_tweaked: Whether the combined pubkey was tweaked
- *          tweak: If is_tweaked, array with the 32-byte tweak
- * internal_key_parity: If is_tweaked, the parity of the combined pubkey
- *                 before tweaking
+/** Opaque data structures
+ *
+ *  The exact representation of data inside is implementation defined and not
+ *  guaranteed to be portable between different platforms or versions. It can,
+ *  however, be safely copied/moved. If you need to convert to a format suitable
```

Comment:
Just to be really safe, is it worth emphasizing even here that the secret nonce should not be copied?

Reply:
Good catch. I removed that sentence and added it instead to the individual doc of the datastructures where applicable.